flow360/component/resource_base.py

```diff
@@ -377,3 +395,184 @@ def from_cloud(cls):
+        Download log file from surface mesh.
+
+        :param to_file: Destination folder path to save the log file(s).
+        :param keep_folder: Flag to indicate whether to keep the folder structure while downloading.
+        :param overwrite: Flag to indicate whether to overwrite existing files.
+        :param progress_callback: Optional callback function for progress updates.
+        :param kwargs: Additional keyword arguments passed to Flow360Resource.download_file().
+        """
+        self.dir_path = to_file
+        file_list = self.flow360.get_download_file_list()
+        regex = re.compile(r"logs/.*\.log")
+        for fname in file_list:
+            if regex.match(fname):
+                self.flow360.download_file(fname, to_file)
+            self.path = to_file + fname
+            self.paths.append(self.path)
+
+    def clean(self):
+        """
+        Clean up the log file and associated folder.
+        """
+        try:
+            for root, dirs, files in os.walk(self.dir_path, topdown=False):
+                for file in files:
+                    file_path = os.path.join(root, file)
+                    os.remove(file_path)
+                for directory in dirs:
+                    dir_path = os.path.join(root, directory)
+                    os.rmdir(dir_path)
+                os.rmdir(self.dir_path)
+            os.rmdir(self.path)
+            self.path = None
+        except FileNotFoundError:
+            self.path = None
+        except PermissionError as error:
+            log.error(f"cannot remove temporary log file under {self.dir_path}", error)
+        except OSError as error:
+            log.error(f"OS error when removing temporary log files {error}")
+
+    def get_log_by_pos(self, pos: Position = None, num_lines: int = 100, fname: str = None):
+        """
+        Get log lines based on position (head, tail, all).
+
+        :param pos: Position enum (HEAD, TAIL, or ALL).
+        :param num_lines: Number of lines to retrieve (for HEAD and TAIL positions).
+        :return: List of log lines.
+        """
+        try:
+            self.download_log()
+            if len(self.paths) >= 1 and fname is not None:
+                self.path = fname
+            with open(self.path, "r", encoding="utf-8") as file:
+                if pos == Position.HEAD:
+                    return file.readlines()[:num_lines]
+                if pos == Position.TAIL:
+                    return file.readlines()[-num_lines:]
+                return file.readlines()[:]
+
+        except (OSError, IOError) as error:
+            log.error("invalid path to log files", error)
+            return None
+
+    def get_log_by_level(self, level: LogLevel = None, fname: str = None):
+        """
+        Get log lines filtered by log level.
+
+        :param level: Log level (ERROR, WARNING, INFO, or None for all).
+        :return: List of filtered log lines.
+        """
+        try:
+            self.download_log()
```

Comment:
without explicit call 'download' can we download to temporary directory https://docs.python.org/3/library/tempfile.html and clean if not explicitly downloaded by user? The sequence of events will be:
```python
case.logs.tail()
# now we download to temp directory and run get_log_by_pos

case.logs.errors()
# now we read already downloaded file and run get_log_by_level

case.logs.to_file('case.log')
# now copy temp file to case.log

# end of scope, remove temp file.
```